### PR TITLE
Contact form: Makes the fields get sent and displayed in correct order

### LINKED ISSUE #49: Contact Form: submission email doesn't match the order of the Form fields
When putting together a contact form using the module bundled with Jetpack I configured the fields to look like.
- Name:
- Email:
- Subject:
- Message:

However, when I receive a form submission, the email shows the following:
- Name:
- Email:
- Message:
- Subject:

Is there anyway to get the submission email fields to match that of the form? Seems backwards that Subject would be placed after the submitters message.

---

The author name, email, and URL, as well as the comment are the <a href="https://github.com/Automattic/jetpack/blob/2.7/modules/contact-form/grunion-contact-form.php#L1086">4 default fields in the feedback form</a>.

Every other field is then added to the Feedback email <a href="https://github.com/Automattic/jetpack/blob/2.7/modules/contact-form/grunion-contact-form.php#L1197">below the default fields</a>. We'll consequently need to create an exception for the "Subject" field.

Suggested here:
- http://wordpress.org/support/topic/contact-form-submission-email-doesnt-match-contact-form-fields?replies=1

### PR DESCRIPTION
**Pull request by zinigor**
This PR introduces a new method that compiles the form into an array of lines that is sorted by field indexes, so that no field is out of place. Also now both the success message that shows the completed form and the email body are generated by a single method. Fixes #49 

Replaces #1433

### CODE REVIEW

File: modules/contact-form/grunion-contact-form.php [hunk 1]
```diff
@@ -1103,9 +1103,32 @@ static function parse( $attributes, $content ) {
 		return $r;
 	}
 
+	/**
+	 * Returns a success message to be returned if the form is sent via AJAX.
+	 *
+	 * @param Integer $feedback_id
```
Comment (dereksmart) [inline modules/contact-form/grunion-contact-form.php]: Can you change to `@param int` for consistency with other docs?

File: modules/contact-form/grunion-contact-form.php [hunk 2]
```diff
@@ -1103,9 +1103,32 @@ static function parse( $attributes, $content ) {
...
+	 * @param Grunion_Contact_Form $form
```
Comment (dereksmart) [inline modules/contact-form/grunion-contact-form.php]: is `Grunion_Contact_Form` an object or array?

File: modules/contact-form/grunion-contact-form.php [hunk 3]
```diff
@@ -1103,9 +1103,32 @@ static function parse( $attributes, $content ) {
...
+	 *
+	 * @return String $message
```
Comment (dereksmart) [inline modules/contact-form/grunion-contact-form.php]: plz lowercase `string` for consistency with others

File: modules/contact-form/grunion-contact-form.php [hunk 4]
```diff
@@ -1103,9 +1103,32 @@ static function parse( $attributes, $content ) {
...
 
+	/**
+	 * Returns a compiled form with labels and values in a form of  an array
+	 * of lines.
+	 * @param Integer $feedback_id
+	 * @param Grunion_Contact_Form $form
+	 *
+	 * @return Array $lines
+	 */
+	static function get_compiled_form( $feedback_id, $form) {
```
Comment (dereksmart) [inline modules/contact-form/grunion-contact-form.php]: Missing whitespace in parenthesis.

**Comment by dereksmart** [comment]
Other than the nitpicks I mentioned, this looks ok and tests well in both Jetpack and WPCOM

**Comment by zinigor** [comment]
Got a :+1: from @jessefriedman, setting as Design Review Complete.

**Pull request merged**